src/functional/ffront/foast_passes/utils.py

```diff
@@ -0,0 +1,19 @@
+from functional.ffront import field_operator_ast as foast
+
+
+def compute_assign_indices(
+    targets: list[foast.FieldSymbol | foast.TupleSymbol | foast.ScalarSymbol | foast.Star],
+) -> list[int | tuple]:
+    indices = list(range(len(targets)))
+    for idx, elt in enumerate(targets):
+        if isinstance(elt, foast.Star):
+            break
+        indices[idx] = idx
+    for idx, elt in reversed(list(enumerate(targets))):
+        rel_idx = idx - len(targets)
+        if isinstance(elt, foast.Star):
+            star_lower, star_upper = max(indices), min(indices)
+            indices[idx] = (star_lower, star_upper)
```

Comment:
I would suggest to place this outside the loop to keep the function seperated into 3 easily to digestible parts.